samples/sample_common/src/vaapi_utils_drm.cpp

```diff
@@ -420,10 +421,45 @@ void* drmRenderer::acquire(mfxMemId mid)
         int ret = m_drmlib.drmIoctl(m_fd, DRM_IOCTL_GEM_OPEN, &flink_open);
         if (ret) return NULL;
 
-        ret = m_drmlib.drmModeAddFB(m_fd,
-              vmid->m_image.width, vmid->m_image.height,
-              24, 32, vmid->m_image.pitches[0],
-              flink_open.handle, &fbhandle);
+        if (VA_FOURCC_NV12 == vmid->m_fourcc) {
+            struct drm_i915_gem_set_tiling set_tiling;
+            memset(&set_tiling, 0, sizeof(set_tiling));
+            set_tiling.handle = flink_open.handle;
+            set_tiling.tiling_mode = I915_TILING_Y;
+            set_tiling.stride = vmid->m_image.pitches[0];
+            ret = m_drmlib.drmIoctl(m_fd, DRM_IOCTL_I915_GEM_SET_TILING, &set_tiling);
+            if (ret) {
+                msdk_printf(MSDK_STRING("DRM_IOCTL_I915_GEM_SET_TILING Failed ret = %d\n"),ret);
+                return NULL;
+            }
+
+            uint32_t handles[4], pitches[4], offsets[4];
+
+            memset(&handles, 0, sizeof(handles));
+            handles[0] = flink_open.handle;
+            handles[1] = flink_open.handle;
+            handles[2] = flink_open.handle;
+
+            memset(&pitches, 0, sizeof(pitches));
+            pitches[0] = vmid->m_image.pitches[0];
+            pitches[1] = vmid->m_image.pitches[1];
+            pitches[2] = vmid->m_image.pitches[2];
+
+            memset(&offsets, 0, sizeof(offsets));
+            offsets[0] = vmid->m_image.offsets[0];
+            offsets[1] = vmid->m_image.offsets[1];
+            offsets[2] = vmid->m_image.offsets[2];
+
+            ret = m_drmlib.drmModeAddFB2(m_fd, vmid->m_image.width, vmid->m_image.height,
+                DRM_FORMAT_NV12, handles, pitches, offsets, &fbhandle, 0);
+        }
+        else
+        {
+            ret = m_drmlib.drmModeAddFB(m_fd,
+            vmid->m_image.width, vmid->m_image.height,
+            24, 32, vmid->m_image.pitches[0],
+            flink_open.handle, &fbhandle);
```

Comment:
@VUNNYSODHI , can we use drmModeAddFB2 for RGB as well?

Reply:
No, for RGB, we have to continue using  drmModeAddFB

Reply:
Okay

Reply:
Ohh, I see Vunny enabled both NV12 and RGB via drmModeAddFB2WithModifiers. Cool!